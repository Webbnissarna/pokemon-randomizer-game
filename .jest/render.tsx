import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
