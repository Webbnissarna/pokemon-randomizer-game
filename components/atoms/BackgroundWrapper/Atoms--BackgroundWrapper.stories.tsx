import { Heading, Paragraph } from "theme-ui";
import Background from ".";

export function Default() {
  return (
    <Background>
      <Heading>Hello world</Heading>
      <Paragraph>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Quos,
        asperiores tenetur! Voluptates quam laudantium et enim harum, eveniet
        perspiciatis ratione nobis explicabo id, ipsam fuga autem incidunt minus
        atque quas.
      </Paragraph>
    </Background>
  );
}
