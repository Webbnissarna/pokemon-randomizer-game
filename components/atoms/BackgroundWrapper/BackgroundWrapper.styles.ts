import { ThemeUIStyleObject } from "theme-ui";

export const mainStyle: ThemeUIStyleObject = {
  background: "background",
  height: "100%",
  width: "100%",
  padding: "md",
};
