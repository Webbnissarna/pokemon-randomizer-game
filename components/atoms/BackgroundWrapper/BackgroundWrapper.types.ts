export interface BackgroundWrapperProps {
  children?: React.ReactNode;
}
