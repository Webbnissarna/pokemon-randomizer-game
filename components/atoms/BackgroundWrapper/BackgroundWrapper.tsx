import { Box } from "theme-ui";
import { mainStyle } from "./BackgroundWrapper.styles";
import type { BackgroundWrapperProps } from "./BackgroundWrapper.types";

export default function BackgroundWrapper({
  children,
}: BackgroundWrapperProps): JSX.Element {
  return <Box sx={mainStyle}>{children}</Box>;
}
