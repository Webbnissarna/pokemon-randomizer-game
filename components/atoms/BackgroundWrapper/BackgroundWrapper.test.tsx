import { render } from "../../../.jest/render";
import BackgroundWrapper from ".";
import { screen } from "@testing-library/react";

describe("Background", () => {
  it("renders", () => {
    const { asFragment } = render(<BackgroundWrapper />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders children", () => {
    const childId = "test-child";
    const childContent = "Hello World";

    render(
      <BackgroundWrapper>
        <span data-testid={childId}>{childContent}</span>
      </BackgroundWrapper>
    );

    const childElement = screen.getByTestId(childId);

    expect(childElement.textContent).toBe(childContent);
  });
});
