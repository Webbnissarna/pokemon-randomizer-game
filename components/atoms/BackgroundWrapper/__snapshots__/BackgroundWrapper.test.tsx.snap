// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-1sw3884-Box"
  />
</DocumentFragment>
`;
